File: src/main/java/com/koddy/server/file/application/usecase/UploadFileUseCase.java
package com.koddy.server.file.application.usecase;

import com.koddy.server.file.application.adapter.FileManager;
import com.koddy.server.file.application.usecase.command.UploadFileCommand;
import com.koddy.server.global.annotation.UseCase;
import lombok.RequiredArgsConstructor;

@UseCase
@RequiredArgsConstructor
public class UploadFileUseCase {
    private final FileManager fileManager;

    public String invoke(final UploadFileCommand command) {
        return fileManager.upload(command.file());
    }
}

File: src/main/java/com/koddy/server/file/application/usecase/UploadFileUseCase.kt
package com.koddy.server.file.application.usecase

import com.koddy.server.file.application.adapter.FileManager
import com.koddy.server.file.application.usecase.command.UploadFileCommand
import com.koddy.server.global.annotation.UseCase

@UseCase
class UploadFileUseCase(
    private val fileManager: FileManager,
) {
    fun invoke(command: UploadFileCommand): String = fileManager.upload(command.file)
}
